### DISCUSSION TITLE
How to use nested Navigation with refreshed?

### DISCUSSION THREAD

**Discussion by HachichaRacem** [Aniketkhote/refreshed#16] [Q&A]
Previously and using Getx 4.6.6, I was easily able to implement nested navigation using the `Navigator` widget provided by flutter with `key: Get.nestedKey(x) // x being int` and then whenever needed I could simply use `Get.toNamed('/....', id: x)` or `Get.back(id:x)` etc..

I have tried to achieve the same with refreshed but I found out that `Get.nestedKey()` now returns `GetDelegate<dynamic>?` and when I tried to use `key : Get.nestedKey("x")?.navigatorKey` it throws `GetRoot is not part of the three` exception.

Could anyone guide me through this?

```dart
void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return GetMaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Navigator(
          key: Get.nestedKey("0")?.navigatorKey, // here
          initialRoute: '/',
          onGenerateRoute: (settings) {
            switch (settings.name) {
              case '/settings':
                return GetPageRoute(
                  page: () => const SettingsScreen(),
                  settings: const RouteSettings(name: '/settings'),
                );
              default:
                return GetPageRoute(
                  page: () => const HomeScreen(),
                  settings: RouteSettings(name: settings.name),
                );
            }
          },
        ),
      ),
    );
  }
}```

**Comment by Aniketkhote** [comment]
Thank you for sharing your experience!

The navigation system in Refreshed is indeed more complex, and I’m still working on understanding and addressing these kinds of issues. The routing and nested navigation methods need better documentation and evaluation to ensure they work as expected.

If you’ve identified specific challenges or have suggestions, it would be incredibly helpful if you could contribute or share more details. Together, we can improve this!
